src/components/app/ProfileFilterNavigator.js

```diff
@@ -44,57 +58,142 @@ class ProfileFilterNavigatorBarImpl extends React.PureComponent<Props> {
+    });
+  };
+
   render() {
     const {
       className,
       items,
       selectedItem,
       uncommittedItem,
       onPop,
-      filterPageData,
       rootRange,
+      filterPageDataForActiveTab,
+      pageDataByTabID,
+      tabFilter,
     } = this.props;
 
     let firstItem;
-    if (filterPageData) {
+    if (filterPageDataForActiveTab) {
+      // TODO: Remove this once we ship the tab selector and remove the active tab view.
       firstItem = (
         <>
-          {filterPageData.favicon ? (
-            <Icon iconUrl={filterPageData.favicon} />
+          {filterPageDataForActiveTab.favicon ? (
+            <Icon iconUrl={filterPageDataForActiveTab.favicon} />
           ) : null}
-          <span title={filterPageData.origin}>
-            {filterPageData.hostname} (
+          <span title={filterPageDataForActiveTab.origin}>
+            {filterPageDataForActiveTab.hostname} (
             {getFormattedTimeLength(rootRange.end - rootRange.start)})
           </span>
         </>
       );
     } else {
-      firstItem = (
-        <Localized
-          id="ProfileFilterNavigator--full-range-with-duration"
-          vars={{
-            fullRangeDuration: getFormattedTimeLength(
-              rootRange.end - rootRange.start
-            ),
-          }}
-        >
-          Full Range
-        </Localized>
-      );
+      // pageDataByTabID will be empty if there is no page information in the
+      // profile or when the page information is empty. This could happen for
+      // older profiles and profiles from external importers that don't have
+      // this information.
+      // FIXME: Change it to false before merging.
+      if (true && pageDataByTabID && pageDataByTabID.size > 0) {
+        const pageData =
+          tabFilter !== null ? pageDataByTabID.get(tabFilter) : null;
+
+        const itemContents = pageData ? (
+          <>
+            {/* Show the page data if the profile is filtered by tab */}
+            {pageData.favicon ? <Icon iconUrl={pageData.favicon} /> : null}
+            <span title={pageData.origin}>
+              {pageData.hostname} (
+              {getFormattedTimeLength(rootRange.end - rootRange.start)})
+            </span>
+          </>
+        ) : (
+          <Localized
+            id="ProfileFilterNavigator--full-range-with-duration"
+            vars={{
+              fullRangeDuration: getFormattedTimeLength(
+                rootRange.end - rootRange.start
+              ),
+            }}
+          >
+            Full Range
+          </Localized>
+        );
+
+        if (items.length === 0 && !uncommittedItem) {
+          // It should be a clickable button if there are no committed ranges.
+          firstItem = (
+            <button
+              type="button"
+              onClick={this._showTabSelectorMenu}
+              className={classNames(
+                'filterNavigatorBarItemContent',
+                'profileFilterNavigator--tab-selector'
```

Comment:
I'm surprised that the menuButtons classes are not reused, but maybe the fact we're in the profile navigator doesn't make it easy.